### ISSUE TITLE
Expo Notification ReceivedListener Wrong Data on Ios

### ISSUE THREAD

**Issue by victorwads** [expo/expo#10070]
https://github.com/expo/expo/blob/78a0d9f1cb47300a01cc099850432c7da808a5e8/packages/expo-notifications/ios/EXNotifications/Notifications/EXNotificationSerializer.m#L54
First I've opened this https://forums.expo.io/t/expo-notification-receivedlistener-wrong-data-on-ios/42469

SDK Version: ^38.0.0
Expo Notifications: ~0.3.3
Platforms: Ios

---------------
# The Problem

On Ios, the method addNotificationReceivedListener on Ios returns a notification with wrong data information.
the following `notification.request.content.data` isn't the same com `Android` and `iOS`

### Android
```javascript
notification.request.content.data.scheduleID  = rDFSDFyrwERTTdfsDsSA
notification.request.content.data.userId  = fqeWtERGfdsSDfs
```
### iOS
```javascript
notification.request.content.data.scheduleID  = null
notification.request.content.data.userId  = null
```

# How reproduce:

## On https://expo.io/notifications
```
body = [{
    to: ['ExponentPushToken[xxxAndroidxxx]','ExponentPushToken[xxxiOSxxx]'],
    title: 'test',
    body: 'test chrome console',
    data: {"scheduleID": "rDFSDFyrwERTTdfsDsSA", "userId": "fqeWtERGfdsSDfs"},
    priority: 'high',
    sound: 'default',
    ttl: 3000,
    channelId: 'default'
}]
fetch('https://exp.host/--/api/v2/push/send', {
    method: 'post',
    body: JSON.stringify(body),
    headers: { 'Content-Type': 'application/json' }
})
```

# On React Native
```
import { addNotificationReceivedListener } from 'expo-notifications'

addNotificationReceivedListener(notification => {
    console.warn(notification)
})

```

### On Android addNotificationReceivedListener receives: [Right]
```
Object {
  "date": 1599202933398,
  "request": Object {
    "content": Object {
      "autoDismiss": true,
      "badge": null,
      "body": "test chrome console",
      "data": Object {
        "scheduleID": "rDFSDFyrwERTTdfsDsSA",
        "userId": "fqeWtERGfdsSDfs",
      },
      "sound": "default",
      "subtitle": null,
      "title": "test",
    },
    "identifier": "0:1599202933404964%0ac519e6f9fd7ecd",
    "trigger": ...
}

```

### On iOS addNotificationReceivedListener receives: [Wrong]
```
Object {
  "date": 1599202932.617644,
  "request": Object {
    "content": Object {
      "attachments": Array [],
      "badge": null,
      "body": "test chrome console",
      "categoryIdentifier": "",
      "data": Object {
        "aps": Object {
          "alert": Object {
            "body": "test chrome console",
            "launch-image": "",
            "subtitle": "",
            "title": "test",
          },
          "category": "",
          "sound": "default",
          "thread-id": "",
        },
        "body": Object {
          "scheduleID": "rDFSDFyrwERTTdfsDsSA",
          "userId": "fqeWtERGfdsSDfs",
        },
      },
      ....
    },
    "identifier": "D5871CD7-8130-4AC6-8465-40CBC20A157F",
    "trigger": ...
}
```

# Where I thing is the solution.

@lukmccall, i think you can help. I don't have a Mackbook, but I've opened the code on VSCode in ubuntu after a long expo cloning and a realize that method `flatDataDictionary` should get each key inside `data[@"body"]` and put though `NSMutableDictionary *result = [data mutableCopy]` but I don't know Objective C. Can you help me improve this?
```
+ (NSDictionary *)serializedNotificationContent:(UNNotificationContent *)content isRemote:(BOOL)isRemote
{
// ...
  NSDictionary *data = isRemote ? [self flatDataDictionary:content.userInfo] : content.userInfo;
  serializedContent[@"data"] = data ?: [NSNull null];
// ...
  return serializedContent;
}

+ (NSDictionary * _Nullable)flatDataDictionary:(NSDictionary * _Nullable)data
{
  if (data[@"body"] && data[@"aps"]) {
    NSMutableDictionary *result = [data mutableCopy];
    [result addEntriesFromDictionary:data[@"body"]];
    [result removeObjectForKey:@"body"];
    return result;

  return data;
}
``` 
https://github.com/expo/expo/blob/78a0d9f1cb47300a01cc099850432c7da808a5e8/packages/expo-notifications/ios/EXNotifications/Notifications/EXNotificationSerializer.m#L45

Working on Typescript workarround

**Comment by lukmccall** [comment]
Hi @victorwads. I believe that this issue was fixed by https://github.com/expo/expo/pull/9164.  You're using `expo-notifications 0.3.3`, and that fix was published in `0.5.0`. If you are using a bare workflow, you can upgrade the package. However, in managed workflow, this change will be available in the next SDK (`39`) .

**Comment by victorwads** [comment]
@lukmccall, thanks I hope that my contribution to expo-notifications could go on SDK 39 too:
https://github.com/expo/expo/pull/10071

It's a work around for both "Bare" and "Managed" on Typescript and also add a Typed Data "feature"

**Comment by bill-pairaktaridis** [comment]
I've basically implemented a workaround for this

`let data = notification.data.body || notification.data`

But it should definitely have been documented somewhere. This has caused huge headaches for our users.

**Comment by victorwads** [comment]
> I've basically implemented a workaround for this
> 
> `let data = notification.data.body || notification.data`
> 
> But it should definitely have been documented somewhere. This has caused huge headaches for our users.

Thanks, it will help while the new version don't comes out.
@bill-pairaktaridis if u can, look this PR: https://github.com/expo/expo/pull/10071

**Comment by bill-pairaktaridis** [comment]
@victorwads I'm not sure I'm the right person to tag for that. I'm not affiliated with Expo.

**Comment by lukmccall** [comment]
I believe that this issue was fixed by #9164. So I'm closing this issue. However, feel free to open another one if this bug still occurs on the newest version.
